src/software/thunderscope/robot_communication.py

```diff
@@ -44,25 +65,116 @@ def __init__(
         except Exception:
             raise Exception("Could not find estop, make sure its plugged in")
 
-    def __enter__(self):
+    def __send_estop_state(self):
+        while True:
+            self.full_system_proto_unix_io.send_proto(
+                EstopState, EstopState(is_playing=self.estop_reader.isEstopPlay())
+            )
+            time.sleep(0.1)
+
+    def run(self):
+        """Forward World and PrimitiveSet protos from fullsystem to the robots.
+
+        If the emergency stop is tripped, the PrimitiveSet will not be sent so
+        that the robots timeout and stop.
+
+        NOTE: If disconnect_fullsystem_from_robots is called, then the packets
+        will not be forwarded to the robots. 
+
+        send_override_primitive_set can be used to send a primitive set, which
+        is useful to dip in and out of robot diagnostics.
+
+        """
+        while True:
+            if self.fullsystem_connected_to_robots:
```

Comment:
I believe from our previous testing we ended up having to initialize `self.fullsystem_connected_to_robots` to False or else the variable wouldn't be initialized before this read